code/game/g_cmds.c

```diff
@@ -5110,6 +5110,58 @@ static void Cmd_MeActionLocal_f(gentity_t* ent)
     }
 }
 
+/*
+=================
+Cmd_RollAction_f
+
+Simple die roll command
+
+/roll [dieSize]
+=================
+*/
+
+// Dice Roll Parameters
+#define ROLL_DIE_MINIMUM_SIDES 2
+#define ROLL_DIE_DEFAULT_SIDES 20
+#define ROLL_DIE_MAXIMUM_SIDES 1000
+
+static void Cmd_RollAction_f(gentity_t* ent)
+{
+    if (!ent->client) {
+        return;
+    }
+
+    // n00b Check
+    if (g_classData[ent->client->sess.sessionClass].isn00b) {
+        trap_SendServerCommand(ent - g_entities, "print \"[n00bs cannot use this command]\n\"");
+        return;
+    }
+
+    int dieMax = ROLL_DIE_DEFAULT_SIDES;
+
+    // If an argument provided, use that as die size
+    if (trap_Argc() > 1) {
+        char arg[MAX_TOKEN_CHARS];
+        trap_Argv(1, arg, sizeof(arg));
+        dieMax = atoi(arg);
+
+        // Check die size within range
+        if (dieMax < ROLL_DIE_MINIMUM_SIDES || dieMax > ROLL_DIE_MAXIMUM_SIDES) {
+            trap_SendServerCommand(ent - g_entities,
+                va("print \"roll: Choose a die size between %d and %d (inclusive)\n\"",
+                    ROLL_DIE_MINIMUM_SIDES, ROLL_DIE_MAXIMUM_SIDES));
+            return;
+        }
+    }
+
+    // "Roll Die"
+    int dieResult = irandom(1, dieMax);
+
+    // Broadcast Result
+    trap_SendServerCommand(-1, va("print \"%s" S_COLOR_WHITE " rolled %u on a d%u\n\"",
```

Comment:
Does this send the result to everyone or just the client who calls roll?

Reply:
Sends it out to everyone, which I expected to be the main use case.

Reply:
Ok, makes sense then to have the n00b check. It would be nice to extract the n00b check to a macro and to define -1 as a more descriptive constant, but those are both separate from this PR.